llap-server/pom.xml

```diff
@@ -38,6 +38,7 @@
       <groupId>org.apache.hive</groupId>
       <artifactId>hive-exec</artifactId>
       <version>${project.version}</version>
+      <classifier>core</classifier>
```

Comment:
don't use the core artifact - that's just bad!

what are you trying to achieve here with this?

Reply:
branch-2.3 ?
please note that changes should land on master first

Reply:
As more dependencies are relocated here, some modules if they depends on non-core artifact, will cause class not found error...

The motivation is because we want to use shaded version of hive-exec (i.e., w/o classifier) in Spark to make sure it doesn't conflict guava version there. But there are more dependencies conflict with Spark. We need to relocate these dependencies too..

Reply:
@sunchao do we need to have similar change on master first?

Reply:
we should fix the issues with using the normal `hive-exec` artifact if there is any - loading the core jar could cause troubles...

Reply:
note: on branch2 guava is most likely not properly shaded away HIVE-22126

Reply:
@kgyrtkirk Guava is shaded in branch-2.3 via https://issues.apache.org/jira/browse/HIVE-23980. The issue is, in order for Spark to use shaded `hive-exec`, Hive will need to relocate more classes and at the same time making sure it won't break other modules (for instance, if the shaded class appears in certain API and another module imported the unshaded version of the class by itself).

Currently we've abandoned this approach and decided to shade the `hive-exec-core` within Spark itself, following similar approach in Trino (see https://github.com/trinodb/trino-hive-apache).

Reply:
we could relocate/shade away those deps to make it possible for other projects to use the normal artifact  - seems like there is a very good list in the trino project.